recorded_future/komand_recorded_future/connection/connection.py

```diff
@@ -14,7 +14,22 @@ def __init__(self):
 
     def connect(self, params):
         self.token = params.get("api_key").get("secretKey")
-        self.client = ApiV2Client(auth=self.token)
+        self.app_version = self.setup_custom_header()
+        self.app_name = "rapid7_insightconnect"
+        self.headers = {
+            "User-Agent": f"{self.app_name}/{self.app_version}",
+            "X-RFToken": self.token
+        }
+
+        self.client = ApiV2Client(auth=self.token, app_name=self.app_name, app_version=self.app_version)
 
     def test(self):
         return demo_test.demo_test(self.token, self.logger)
+
+    def setup_custom_header(self):
+        with open("../../plugin.spec.yaml") as f:
+            spec = yaml.load(f)
+
+        return spec.get("version")
```

Comment:
You should be able to get this information from the runtime directly

Reply:
`self.connection.meta.version`

Reply:
I looked for that via the debug inspector and it wasn't available.

Reply:
Fixed (sort of)